src/caikit_nlp_client/http_client.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from typing import Any
+
+import requests
+
+from src.caikit_nlp_client.http_config import HTTPConfig
+
+log = logging.getLogger(__name__)
+
+
+class HTTPCaikitNlpClient:
+    """HTTP client for a caikit nlp runtime server
+
+    Args:
+        http_config (HTTPConfig): Configurations to make HTTP call.
+    """
+
+    def __init__(self, http_config: HTTPConfig) -> None:
+        mode = "http"
+        if http_config.mtls or http_config.tls:
+            mode = "https"
```

Comment:
The ternary operator might be useful:
```suggestion
        protocol = "https" if (http_config.mtls or http_config.tls) else mode = "http"
```

Reply:
done